src/runtime/api-channel.ts

```diff
@@ -451,6 +455,16 @@ function AutoConstruct<S extends {prototype: {}}>(target: S) {
               args[a.position] = (args[a.position] as (() => unknown))();
             });
           }
+
+          // For redundant initializers, we first check if we already have a
+          // mapping for that ID.
+          if (initializer !== -1 && descriptor[initializer].redundant) {
+            if (this._mapper.hasMappingForId(messageBody['identifier'])) {
```

Comment:
This might break hot reloading and/or reinstantiating particles. It would be worth doing some manual checks on the logic around `reinstantiate()` and `reload()` in `ParticleExecutionHost`.